torch/distributed/fsdp/_runtime_utils.py

```diff
@@ -347,9 +347,12 @@ def _reshard(
     """
     handle.reshard(free_unsharded_flat_param)
     if state.limit_all_gathers and free_unsharded_flat_param:
-        free_event = state._device_handle.Event()
-        free_event.record()
-        state._free_event_queue.enqueue(free_event)
+        if not torch.distributed._functional_collectives.is_torchdynamo_compiling():
+            # We don't run a even queue for freeing under torch compile atm
+            # But maybe we need to? TODO(voz): Look into this
```

Comment:
big TODO, not blocking atm

Reply:
i think its fine to diverge here, insofar as compile captures a coherent program that it can optimize.  e.g. I don't care if compile uses the same sets of streams/events or if it totally ignores eager prefetching and does its own prefetching.

but I care that compiling fsdp doesn't fall on its face if we turn prefetching on and some other code hangs waiting for an event that will never be enqueued.

Reply:
Agreed, I can test this post MVP - my current goal is to get the least amount of things needed to land an MVP. 

The plan is to actually support this, so the divergence should be short lived.

Reply:
alright, well if @awgu is Ok with this one then I am

Reply:
Why not just run the MVP under `limit_all_gathers == False` so that you don't hit these queues at all?